gef-ui/server/controllers/application-submission/index.js

```diff
@@ -1,21 +1,27 @@
 import * as api from '../../services/api';
 import { validationErrorHandler } from '../../utils/helpers';
 import { PROGRESS } from '../../../constants';
+import Application from '../../models/application';
 
 const maxCommentLength = 400;
 
-export const getApplicationSubmission = (req, res) => {
-  const { params } = req;
+export const getApplicationSubmission = async (req, res) => {
+  const { params, session } = req;
   const { applicationId } = params;
-  return res.render('application-details-comments.njk', { applicationId, maxCommentLength });
+  const { user, userToken } = session;
+  const application = await Application.findById(applicationId, user, userToken);
+  const isAutomaticCover = application.coverTerms && application.coverTerms.isAutomaticCover;
+
+  return res.render('application-details-comments.njk', { applicationId, isAutomaticCover, maxCommentLength });
 };
 
 export const postApplicationSubmission = async (req, res, next) => {
-  const { params, body } = req;
-  const { userToken } = req.session;
+  const { params, session, body } = req;
+  const { user, userToken } = session;
   const { applicationId } = params;
   const { comment } = body;
-  const application = await api.getApplication(applicationId);
+  const application = await Application.findById(applicationId, user, userToken);
+  const isAutomaticCover = application.coverTerms && application.coverTerms.isAutomaticCover;
```

Comment:
Maybe worth using `submissionType` with constants to get the `isAutomatic` boolean. I suspect this approach is used elsewhere though. Food for thought